crates/forge/tests/cli/zk/script.rs

```diff
@@ -108,3 +108,80 @@ contract DeployScript is Script {
     let transactions = transactions.transactions;
     assert_eq!(transactions.len(), 3);
 });
+
+forgetest_async!(test_zk_broadcast_raw_create2_deployer, |prj, cmd| {
+    foundry_test_utils::util::initialize(prj.root());
+    let node = ZkSyncNode::start().await;
+    let url = node.url();
+
+    let (_, private_key) = ZkSyncNode::rich_wallets()
+        .next()
+        .map(|(addr, pk, _)| (addr, pk))
+        .expect("No rich wallets available");
+
+    prj.add_source(
+        "Counter.sol",
+        r#"
+    pragma solidity ^0.8.0;
+
+    contract Counter {
+        uint256 public count;
+        function increment() external {
+            count++;
+        }
+    }
+    "#,
+    )
+    .unwrap();
+
+    //deploy
+    let output = cmd.args([
+        "create",
+        "src/Counter.sol:Counter",
+        "--zksync",
+        "--private-key",
+        private_key,
+        "--rpc-url",
+        &url,
+    ]);
+    let address = output.assert_success().get_output().stdout_lossy();
+    println!("2. address: {:?}", address.trim());
+    cmd.forge_fuse();
+
+    prj.add_script(
+        "Foo",
+        r#"
+import "forge-std/Script.sol";
+import {Counter} from "../src/Counter.sol";
+contract SimpleScript is Script {
+    function run() external {
+        // zk raw transaction
+        vm.startBroadcast();
+        // This raw transaction comes from cast mktx of increment() to Counter contract
```

Comment:
Not reproducible, instead use something like this:
```suggestion
        // `cast mktx <ADDRESS> "increment()" ...`
```

Also, from the format it looks like the address is known ahead of time, so why not put it in a constant? (or reuse what we have?) You can even use `create2` to ensure a deterministic deployment

Reply:
Or even bytecode that we already embed elsewhere so we don't get different bytecode with different compiler versions